vaadin-dashboard-flow-parent/vaadin-dashboard-flow/src/main/java/com/vaadin/flow/component/dashboard/DashboardWidget.java

```diff
@@ -22,4 +22,23 @@
 @JsModule("@vaadin/dashboard/src/vaadin-dashboard-widget.js")
 // @NpmPackage(value = "@vaadin/dashboard", version = "24.6.0-alpha0")
 public class DashboardWidget extends Component {
+
+    /**
+     * Returns the title of the widget.
+     *
+     * @return the {@code widget-title} property from the web component
+     */
+    public String getTitle() {
+        return getElement().getAttribute("widget-title");
+    }
+
+    /**
+     * Sets the title of the widget.
+     *
+     * @param title
+     *            the title to set
+     */
+    public void setTitle(String title) {
+        getElement().setAttribute("widget-title", title == null ? "" : title);
+    }
```

Comment:
Looks like title component / the slot way of things is missing compared to the web component

Reply:
We are going to render the title based on the hierarchy automatically on the web component. Therefore, there will not be a component based title for the Flow counterpart for now. However, if the title is empty, the header title section will be of 0 width and the header content section will be filling that space. It will be possible to add any custom title component in that section.